10_datastructures_lab/01.py

```diff
@@ -1,10 +1,24 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+#host to ip address
+import sys
+
+
+addresses = {}
+
+# create a set of unique computer name from argv
+# (excluding program name)
+hostnames = set(sys.argv[1:])
+
+#Read host file and save the host & I.P address in the dictionary
+with open("hosts", "r") as fin:
+    for line in fin:
+        hostaddress = line.strip('\n').split('=')
+        (host, address) = hostaddress
+        addresses[host] = address
+
+#Go over the given names in the argument line and print the I.P address
+for name in hostnames:
+    if name in addresses:
+        print "The I.P. address of %s is %s" % (name, addresses[name])
+    else:
+        print "%s doesn't exist in the hosts file" % (name)
+

```

Comment:
Nice. The only thing I can add is that you don't need the hostaddress variable. You can merge lines 14 and 15 into one statement.